client/config_test.go

```diff
@@ -0,0 +1,274 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"istio.io/manager/apiserver"
+	"istio.io/manager/model"
+)
+
+type FakeHandler struct {
+	expResponse   string
+	expHeaders    http.Header
+	expStatusCode int
+}
+
+func (f *FakeHandler) HandlerFunc(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(f.expStatusCode)
+	for k, v := range f.expHeaders {
+		w.Header().Set(k, v[0])
+	}
+	w.Write([]byte(f.expResponse))
+}
+
+var (
+	errInvalidSpec = errors.New("cannot parse proto message: json: " +
```

Comment:
Testing for specific error strings seems fragile and it looks like the HTTP status codes, which are canonical, should be sufficient.